Sources/Redis/Application+Redis.swift

```diff
@@ -0,0 +1,74 @@
+import Vapor
+
+extension Application {
+    public var redis: Redis {
+        .init(application: self)
+    }
+
+    public struct Redis {
+        struct ConfigurationKey: StorageKey {
+            typealias Value = RedisConfiguration
+        }
+
+        public var configuration: RedisConfiguration? {
+            get {
+                self.application.storage[ConfigurationKey.self]
+            }
+            nonmutating set {
+                if self.application.storage.contains(PoolKey.self) {
+                    fatalError("Cannot modify your Redis configuration after redis has been used")
+                }
+                self.application.storage[ConfigurationKey.self] = newValue
+            }
+        }
+
+        struct PoolKey: StorageKey, LockKey {
+            typealias Value = EventLoopGroupConnectionPool<RedisConnectionSource>
+        }
+
+        internal var pool: EventLoopGroupConnectionPool<RedisConnectionSource> {
+            if let existing = self.application.storage[PoolKey.self] {
+                return existing
+            } else {
+                let lock = self.application.locks.lock(for: PoolKey.self)
+                lock.lock()
+                defer { lock.unlock() }
+                guard let configuration = self.configuration else {
+                    fatalError("Redis not configured. Use app.redis.configuration = ...")
+                }
+                let new = EventLoopGroupConnectionPool(
+                    source: RedisConnectionSource(configuration: configuration, logger: self.application.logger),
```

Comment:
The `logger` in this argument is which `Logger` will handle logs generated by the `RedisConnectionSource` - do we _not_ want to use the [default value](https://github.com/vapor/redis-kit/blob/master/Sources/RedisKit/RedisConnectionSource.swift#L14)?

Reply:
@Mordil application's logger may contain useful context in the future. If there's a situation where > 1 application is running and outputting data (common in testing), having the context could be helpful in debugging.